connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/InboundConnectorDefinitionImpl.java

```diff
@@ -1,79 +1,75 @@
+    String type,
+    String tenantId,
+    String deduplicationId,
+    @JsonIgnore Map<String, String> rawPropertiesWithoutKeywords,
+    List<InboundConnectorElementImpl> elements
+) implements InboundConnectorDefinition {
+  public InboundConnectorDefinitionImpl(List<InboundConnectorElementImpl> definitions) {
+    this(
+        extractType(definitions),
+        extractTenantId(definitions),
+        extractDeduplicationId(definitions),
+        extractRawProperties(definitions),
+        definitions
+    );
   }
 
-  public String resultExpression() {
-    return rawProperties.get(Keywords.RESULT_EXPRESSION_KEYWORD);
+  private static String extractType(List<InboundConnectorElementImpl> definitions) {
+    if (definitions.stream()
+        .map(InboundConnectorElementImpl::type)
+        .distinct()
+        .count()
+        > 1) {
+      throw new IllegalArgumentException(
+          "All elements in a group must have the same type");
+    }
+    return definitions.getFirst().type();
   }
 
-  public String resultVariable() {
-    return rawProperties.get(Keywords.RESULT_VARIABLE_KEYWORD);
+  private static String extractTenantId(List<InboundConnectorElementImpl> definitions) {
+    if (definitions.stream()
+        .map(InboundConnectorElementImpl::tenantId)
+        .distinct()
+        .count()
+        > 1) {
+      throw new IllegalArgumentException(
+          "All elements in a group must have the same tenant ID");
+    }
+    return definitions.getFirst().tenantId();
   }
 
-  public String activationCondition() {
-    return Optional.ofNullable(rawProperties.get(Keywords.ACTIVATION_CONDITION_KEYWORD))
-        .orElseGet(() -> rawProperties.get(Keywords.DEPRECATED_ACTIVATION_CONDITION_KEYWORD));
+  private static String extractDeduplicationId(List<InboundConnectorElementImpl> definitions) {
+    if (definitions.stream()
+        .map(InboundConnectorElementImpl::deduplicationId)
+        .distinct()
+        .count()
+        > 1) {
+      throw new IllegalArgumentException(
+          "All elements in a group must have the same deduplication ID");
+    }
+    return definitions.getFirst().deduplicationId();
   }
 
-  // override to exclude rawProperties
-  @Override
-  public String toString() {
-    return "InboundConnectorDefinitionImpl{"
-        + "correlationPoint="
-        + correlationPoint
-        + ", bpmnProcessId='"
-        + bpmnProcessId
-        + '\''
-        + ", version="
-        + version
-        + ", processDefinitionKey="
-        + processDefinitionKey
-        + ", elementId='"
-        + elementId
-        + ", tenantId='"
-        + tenantId
-        + '\''
-        + '}';
+  private static Map<String, String> extractRawProperties(List<InboundConnectorElementImpl> definitions) {
+    if (definitions.stream()
+        .map(InboundConnectorElementImpl::rawPropertiesWithoutKeywords)
+        .distinct()
+        .count()
+        > 1) {
```

Comment:
if the `deduplicationMode = MANUAL`, isn't it possible that we have different properties even if the `deduplicationId` is the same?

Reply:
This could even be the case if `deduplicationMode = AUTO`. The correlation id and the message id expression could be different for each element.

Reply:
This wouldn't make sense because we are only starting one executable with a specific set of properties for all connectors with the same `deduplicationId`. If properties are different, which ones should be passed to the executable?

Reply:
Result expression and other runtime-specific properties are out of scope here as we already filtered them out by this stage. So they can be different for sure, but not the user-level properties that are consumed by the connector.